### PR TITLE
fix: 🐛 Fix radio card background color in dark mode

### PR DESCRIPTION
**Pull request by ZedLi**
✅ Closes: ICU-4021

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-4021)

## Description

The background color for the radio cards was also set to white, even for dark mode. 
Also looks like we had an unncessary `display` field

:technologist: [Admin preview](https://boundary-8tg3crtmr-hashicorp.vercel.app/)

:rose: [Rose preview](https://boundary-ui-storybook-fu9l1o0o1-hashicorp.vercel.app/)

### Screenshots (if appropriate):

Before:
image

After:
image

Still looks normal in light mode:
image

### CODE REVIEW

File: addons/rose/addon/styles/rose/components/form/_radio-card.scss [hunk 1]
```diff
@@ -75,7 +75,6 @@ $radio-svg: (
   }
   .rose-form-radio-label {
     @include type.type(s, bold);
-    display: flex;
```
Comment (ZedLi) [inline addons/rose/addon/styles/rose/components/form/_radio-card.scss:78]: It looks like this `display` was always getting overriden by another `display: block` on line 82

**Review by randallmorey — CHANGES_REQUESTED** [review]
What about the title/label?  And is the paragraph text too dark?

File: addons/rose/addon/styles/rose/components/form/_radio-card.scss [hunk 2]
```diff
@@ -75,7 +75,6 @@ $radio-svg: (
...
     color: var(--black);
```
Comment (randallmorey) [inline addons/rose/addon/styles/rose/components/form/_radio-card.scss:78]: We shouldn't use absolute colors like this except in very specific cases.  Use relative poles instead (stark/subtle).

**Review by randallmorey — CHANGES_REQUESTED** [review]

File: addons/rose/addon/styles/rose/components/form/_radio-card.scss (hunk 2 above)
Comment (ZedLi) [reply addons/rose/addon/styles/rose/components/form/_radio-card.scss]: Wow my eyes glazed right over that, I removed it and it looks like it defaults to the body's color of `var(--stark)`, should this be explicit for the `color` and `background-color` or is it ok to inherit it?

Comment (randallmorey) [reply addons/rose/addon/styles/rose/components/form/_radio-card.scss]: I prefer inherited.  Less code.

**Review by calcaide — APPROVED** [review]
LGTM! Nice work

**Review by DhariniJeeva — APPROVED** [review]
LGTM!

File: addons/rose/addon/styles/rose/variables/color/_css.scss [hunk 3]
```diff
@@ -123,9 +123,9 @@
   --ui-border-subtler-1: #{color(ui-gray, 800)};
   --ui-border: #{color(ui-gray, 700)};
 
-  --ui-gray-subtler-6: #{color(ui-gray, 900)};
-  --ui-gray-subtler-5: #{color(ui-gray, 800)};
-  --ui-gray-subtler-4: #{color(ui-gray, 700)};
+  --ui-gray-subtler-6: #{color(ui-gray, 950)};
+  --ui-gray-subtler-5: #{color(ui-gray, 900)};
+  --ui-gray-subtler-4: #{color(ui-gray, 800)};
```
Comment (ZedLi) [inline addons/rose/addon/styles/rose/variables/color/_css.scss:128]: I only made these ones darker in dark mode, I couldn't find good examples of the lighter ones being used so left them as is. Let me know if you think I should also adjust them.

**Review by cameronperera — APPROVED** [review]
LGTM

**Pull request merged**